api/metric/sdkhelpers.go

```diff
@@ -159,21 +160,13 @@ func insertResource(impl MeterImpl, opts []Option) []Option {
 
 // WrapMeterImpl constructs a `Meter` implementation from a
 // `MeterImpl` implementation.
-func WrapMeterImpl(impl MeterImpl) Meter {
+func WrapMeterImpl(impl MeterImpl, libraryName string) Meter {
```

Comment:
The [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-metrics.md#meter-interface) says (in 3rd paragraph):

```
As an obligatory step, the API requires the caller to provide the name
of the instrumenting library (optionally, the version) when obtaining
a `Meter` implementation, that is meant to be used for identifying
instrumentation produced from that library for such purposes as
disabling instrumentation, configuring aggregation, and applying
sampling policies.  (TODO: refer to the semantic convention on the
reporting library name).
```

I'm a bit confused - if an application creates a meter, should it pass a name of the instrumenting library it's going to use? Then the instrumenting library could do it itself, no? So maybe SDK should provide a `NewMeter` function that does `api.WrapMeterImpl(New(…), fmt.Sprintf("go.opentelemetry.io/otel/sdk:v%s", sdk.Version()))`?

Reply:
`WrapMeterImpl` will be used to implement a `Provider`. The Provider will be given a name in the call to `Provider.Meter(string)` and use `WrapMeterImpl` to build the `Meter` from its underlying `MeterImpl` and the name. 

For the mock testing environment, I did provide a `NewMeter()` helper, but it uses a fixed string and there's a mock Provider available if the name is meant for testing.

FYI `fmt.Sprintf("go.opentelemetry.io/otel/sdk:v%s", sdk.Version())` is the "telemetry library", not the "instrumenting library". This has been debated, e.g., 
https://github.com/open-telemetry/oteps/pull/84
https://github.com/open-telemetry/opentelemetry-specification/pull/494